MariaDB/server
# MDEV-25152 "mysqldump --extended-insert" could insert linebreaks between rows

- [x] *The Jira issue number for this PR is: MDEV-25152*


## Description
Currently, *mysqldump --extended-insert* outputs all rows on a single line, which makes it difficult to use with various Unix command-line utilities such as `grep`. This is a small improvement that makes mysqldump emit a linebreak between each row, so that it becomes easier to work with the dumped file. This change has no significant impact on the performance of dumping and restoring and it does not affect compatibility.

## How can this PR be tested?
The test consists on dumping a database with two tables - students and courses - where each table may have one of four scenarios:

**a)** The table is empty - in that case, data dumping has no INSERT clauses
**b)** The table has one single row - in that case, data dumping has one INSERT clause terminated by semicolon;
**c)** The table has 10 rows - in that case, data dumping has one INSERT clause spanned through several lines where the first line has the INSERT keyword and ends by comma;
**d)** The table has 20000 rows - in that case, data dumping has more than one INSERT clause (likely two), each of them spanned through several lines where the first line has the INSERT keyword and ends by comma;

The test combines the four possible scenarios for each one of the two tables so that it generates 16 different databases. Each one of these databases is exported and every INSERT clause in the resulting output is checked against the expected behavior for its table scenario. The database is then imported again to confirm that the generated output is correct.

Attached is a ZIP file with the following files:
`test_issue_25152.py` - Python script to generate the creation script of a database with x students and y courses, where x and y are given as command line parameters.
`test_issue_25152.sh` - Bash script that uses the above Python script to automatize testing for different scenarios
`test_issue_25152.log` - Output log of the test

**NB:** The test scripts are set for a database with no root password (they were based on a local database with completely fictitious data).

## Basing the PR against the correct MariaDB version


## Backward compatibility
Not changed

[mdev-25152.zip](https://github.com/MariaDB/server/files/6697699/mdev-25152.zip)
